extensions/microsoft-authentication/src/AADHelper.ts

```diff
@@ -706,7 +712,7 @@ export class AzureActiveDirectoryService {
 			});
 
 			const proxyEndpoints: { [providerId: string]: string } | undefined = await vscode.commands.executeCommand('workbench.getCodeExchangeProxyEndpoints');
-			const endpointUrl = proxyEndpoints?.microsoft || loginEndpointUrl;
+			const endpointUrl = proxyEndpoints?.microsoft || this._loginEndpointUrl;
 			const endpoint = `${endpointUrl}${scopeData.tenant}/oauth2/v2.0/token`;
```

Comment:
this is where we proxy the request over to vscode.dev.... vscode.dev implements the `'workbench.getCodeExchangeProxyEndpoints'` command. Since we aren't supporting the web, we need to handle this somehow.

Reply:
I think it's worth trying at least the national clouds. While the login endpoints are different for Mooncake and Fairfax, both are exposed to the internet. Custom clouds, it's ultimately up to the owner if they want it to be internet-facing or not.

Reply:
Couple things:
* since `'workbench.getCodeExchangeProxyEndpoints'` is implemented by vscode.dev, vscode.dev actually only uses the Public login url... so using the proxy result here would be useless. For the `azure-cloud` auth provider, we should just not use the proxy for now and instead always use the `this._loginEndpointUrl`
* By using the `this._loginEndpointUrl` always, we will expose the problem that we can't make a request to the identity API due to CORS... but only when this extension is running in a web worker extension host. So, basically we need _some_ logic _somewhere_ to say that that auth provider doesn't work in the web (well in vscode.dev and github.dev basically). I'm thinking maybe `createSession` could throw when we are running in a webworker... and easy way to test for this is to check `typeof navigator === 'undefined'` which will be `true` only in node. ([example](https://github.com/microsoft/vscode/blob/f31dffc7f9b93a1cdef21243ec30dea7d489a615/extensions/github-authentication/src/githubServer.ts#L171-L181))